expr/func_offsettozero_test.go

```diff
@@ -0,0 +1,236 @@
+package expr
+
+import (
+	"math"
+	"strconv"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/test"
+)
+
+func TestOffsetToZero(t *testing.T) {
+	const (
+		positiveOffset = 10
+		negativeOffset = -10
+	)
+	for _, tc := range []struct {
+		name   string
+		input  []models.Series
+		output []models.Series
+	}{
+		{
+			name:   "no input",
+			input:  []models.Series{},
+			output: []models.Series{},
+		},
+		{
+			name: "single positive offset",
+			input: []models.Series{
+				getSeriesNamed("a", []schema.Point{
+					{Val: 0 + positiveOffset, Ts: 10},
+					{Val: 0 + positiveOffset, Ts: 20},
+					{Val: 5.5 + positiveOffset, Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.Inf(1), Ts: 50},
+					{Val: 1234567890 + positiveOffset, Ts: 60},
+				}),
+			},
+			output: []models.Series{
+				getSeriesNamed("offsetToZero(a)", []schema.Point{
+					{Val: 0, Ts: 10},
+					{Val: 0, Ts: 20},
+					{Val: 5.5, Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.Inf(1), Ts: 50},
+					{Val: 1234567890, Ts: 60},
+				}),
+			},
+		},
+		{
+			name: "single negative offset",
+			input: []models.Series{
+				getSeriesNamed("a", []schema.Point{
+					{Val: 0 + negativeOffset, Ts: 10},
+					{Val: 0 + negativeOffset, Ts: 20},
+					{Val: 5.5 + negativeOffset, Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.Inf(1), Ts: 50},
+					{Val: 1234567890 + negativeOffset, Ts: 60},
+				}),
+			},
+			output: []models.Series{
+				getSeriesNamed("offsetToZero(a)", []schema.Point{
+					{Val: 0, Ts: 10},
+					{Val: 0, Ts: 20},
+					{Val: 5.5, Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.Inf(1), Ts: 50},
+					{Val: 1234567890, Ts: 60},
+				}),
+			},
+		},
+		{
+			name: "negative infinity",
+			input: []models.Series{
+				getSeriesNamed("a", []schema.Point{
+					{Val: 0, Ts: 10},
+					{Val: 0, Ts: 20},
+					{Val: 5.5, Ts: 30},
+					{Val: math.Inf(-1), Ts: 40},
+					{Val: math.Inf(1), Ts: 50},
+					{Val: 1234567890, Ts: 60},
+				}),
+			},
+			output: []models.Series{
+				getSeriesNamed("offsetToZero(a)", []schema.Point{
+					{Val: math.Inf(1), Ts: 10},
+					{Val: math.Inf(1), Ts: 20},
+					{Val: math.Inf(1), Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.Inf(1), Ts: 50},
+					{Val: math.Inf(1), Ts: 60},
+				}),
+			},
+		},
+		{
+			name: "multiple",
+			input: []models.Series{
+				getSeriesNamed("a", []schema.Point{
+					{Val: 0 + positiveOffset, Ts: 10},
+					{Val: 0 + positiveOffset, Ts: 20},
+					{Val: 5.5 + positiveOffset, Ts: 30},
+					{Val: math.NaN(), Ts: 40},
+					{Val: math.NaN(), Ts: 50},
+					{Val: 1234567890 + positiveOffset, Ts: 60},
+				}),
+				getSeriesNamed("b", []schema.Point{
```

Comment:
maybe use different numbers for `b` to make the test more specific and robust? (e.g. to check that a and b don't get mixed up)

Reply:
Makes sense, done in https://github.com/grafana/metrictank/pull/1995/commits/c3cce88d1c25917635e84a9e8345366bd88f9e1e